### PR TITLE
Nomenclature for the `SemigroupK` hierarchy

### PR DESCRIPTION
**Pull request by TimWSpence**

### CODE REVIEW

File: docs/nomenclature.md [hunk 1]
```diff
@@ -138,6 +138,30 @@ Like the previous section, we use the `E` for the error parameter type.
...
+| `IterableOnce[F[A]] => Option[F[A]]` | `combineAllOptionK`
+
+### MonoidK
+| Type         | Method Name  | Constraints |
+|------------|--------------|-----------|
+| `F[A]` | `empty`
+| `F[A] => Boolean` | `isEmpty`
+| `IterableOnce[F[A]] => F[A]` | `combineAllK`
+
+### Alternative
```
Comment (satorg) [inline docs/nomenclature.md:156]: Honestly, I've never put attention to this page and thus I've just realized it exists! 😊 
Would you consider mention  about `NonEmptyAlternative` there as well?

| Type | Method Name | Constraints |
| --- | --- | --- |
| `A => F[A] => F[A]` | `prependK` | |
| `F[A] => A => F[A]` | `appendK` | |

Comment (TimWSpence) [reply docs/nomenclature.md:156]: Oh yeah sure!

**Comment by TimWSpence** [comment]
@satorg sorry for the delay! `NonEmptyAlternative` is added now

**Review by satorg — APPROVED** [review]
Nice, thank you!

**Review by danicheg — APPROVED** [review]

**Pull request merged**